Add RosHostname to node executor.

tulku opened the pull request:
Currently the App developer can access the RosMarterUri in a RosActivity. This PR adds a similar method to get the currently defined RosHostname.

To set the hostname the master chooser was modified to now list the available (and UP) network interfaces.

I made a simple modifications to the pubsub example to show how the RosHostname can be used in the applications.

This PR was mainly motivated because of the need to connect to an Android based robot through a VPN, and we needed to force the app to use the tunnel interface instead of the wlan one.

In order for this to work, the following PR in rosjava_core needs to be merged:
https://github.com/rosjava/rosjava_core/pull/177

I make this pull request to the hydro branch because it was stated in a mail that this is the one to use now, but it doesn't seem right. I guess this should be merged to the not yet created indigo-devel branch.

tulku committed:
List all network interfaces in master chooser

A list of detected network interfaces can be found in the advanced
options segment of the master chooser UI. Selecting an interface allows
the user to configure the ROS Hostname. If no interface is selected then
the code fall-back to the previous behaviour.

I made a simple modifications to the pubsub example to show how the
RosHostname can be used in the applications.

chadrockey commented:
Bump.  @damonkohler @stonier

Commit:
upping maven version number

damonkohler committed:
Upgrades Gradle Wrapper to version 1.8.
Removes redundant specification in .gitignore file.

damonkohler committed:
Moves most attributes from AndroidManifest.xml into build.gradle.
Changes android_honeycomb_mr2 API level to 15. Less than 0.1% of devices run level 13.

damonkohler committed:
Renames android_gingerbread_mr1 to android_10.

damonkohler committed:
Renames android_honeycomb_mr2 to android_15.

damonkohler committed:
Changes all Android projects under android_core to support both Eclipse and Android Studio.

damonkohler commented:
Please change this to apply to the master branch. Also, have you signed the CLA?

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -63,8 +73,36 @@
+
+  private class StableArrayAdapter extends ArrayAdapter<String> {
+
+    HashMap<String, Integer> mIdMap = new HashMap<String, Integer>();
```
We're not using the m member prefix.

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -76,6 +114,39 @@ protected void onCreate(Bundle savedInstanceState) {
     uriText.setText(masterUri);
+
+    interfacesList = (ListView) findViewById(R.id.interfacesView);
+    final ArrayList<String> list = new ArrayList<String>();
```
Program to the interface. List&lt;String>

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -76,6 +114,39 @@ protected void onCreate(Bundle savedInstanceState) {
+    final ArrayList<String> list = new ArrayList<String>();
+
+    try {
+      Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
```
for (NetworkInterface networkInterface : Collections.list(NetworkInterface.getNetworkInterfaces())) { ... }

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -76,6 +114,39 @@ protected void onCreate(Bundle savedInstanceState) {
+      public void onItemClick(AdapterView<?> parent, View view,
+                              int position, long id) {
+        selectedInterface = parent.getItemAtPosition(position).toString();
+        Toast.makeText(getApplicationContext(), "Selected: " + selectedInterface,
```
This looks like debugging code.

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -139,31 +208,30 @@ public void qrCodeButtonClicked(View unused) {
-  public Intent createNewMasterIntent (Boolean isPrivate) {
+  public Intent createNewMasterIntent(Boolean newMaster, Boolean isPrivate) {
     Intent intent = new Intent();
-    intent.putExtra("NEW_MASTER", true);
```
These extras should all start with ROS_MASTER_

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/NodeMainExecutorService.java:
```diff
@@ -201,6 +204,13 @@ public void setMasterUri(URI uri) {
     masterUri = uri;
   }
 
+  public void setRosHostname(String hostname) {
```
I'd rather see startMaster() simply wrap the RosCore methods (i.e. newPublicMaster(), newPrivateMaster(), etc.). Additionally, this should happen in a separate thread so that the UI is definitely not blocked. I see you're doing this in the activity. But, if you're doing to do it thee, you may as well not provide a startMaster method here and instead use RosCore directly in the activity.

tulku replied:
I guess that we could actually remove all the startMaster, getMasterUri and the setRosHostname functionality from NodeMainExecutorService and add it directly to the RosActivity. What do you think?

damonkohler replied:
No, it needs to be in the service. I was wrong before about moving it into the activity. You should only ever have one master and there may be multiple RosActivities.

tulku replied:
So what's need to be done is moving the AsyncTask from the ROS Activity over to the service? startMaster() is already just a wrapper over RosCore methods, so I think the method is ok.

damonkohler replied:
SGTM

damonkohler committed:
Renames Camera to XYOrthographicCamera.
Fixes camera control gestures that stopped working after Honeycomb.

tulku committed:
Style improvements

stonier committed:
Merge pull request #176 from damonkohler/upgrade_gradle

Upgrades Gradle Wrapper to version 1.8.

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -63,8 +74,36 @@
+    public boolean hasStableIds() {
+      return true;
+    }
+
```
No blank line at the end of classes.

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -76,6 +115,35 @@ protected void onCreate(Bundle savedInstanceState) {
+      throw new RosRuntimeException(e);
+    }
+
+    // Fallback to previous behaviour when no interface
```
I don't think this needs to be wrapped. Check your line wrapping settings in your IDE. Please wrap consistently at 100 characters. Below you've wrapped things at odd columns.

damonkohler commented on android_gingerbread_mr1/src/main/java/org/ros/android/MasterChooser.java:
```diff
@@ -139,31 +205,30 @@ public void qrCodeButtonClicked(View unused) {
   }
 
-  public Intent createNewMasterIntent (Boolean isPrivate) {
+  public Intent createNewMasterIntent(Boolean newMaster, Boolean isPrivate) {
```
Don't use the uppercase types unless necessary. Use boolean instead.

damonkohler commented on android_gingerbread_mr1/src/main/res/layout/master_chooser.xml:
```diff
@@ -64,8 +64,32 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:visibility="gone"
+            android:id="@+id/advanceOptions">
```
advancedOptions

damonkohler commented on android_gingerbread_mr1/src/main/res/layout/master_chooser.xml:
```diff
@@ -64,8 +64,32 @@
+        <ListView
+                android:layout_width="wrap_content"
+                android:layout_height="0dp"
+                android:id="@+id/interfacesView"
```
Call it networkInterfaces instead?

damonkohler commented on android_gingerbread_mr1/src/main/res/layout/master_chooser.xml:
```diff
@@ -64,8 +64,32 @@
             android:layout_width="fill_parent"
-            android:layout_height="fill_parent">
+            android:layout_height="fill_parent"
+            >
```
Check your autoformat settings for XML. Make sure the style is consistent.

damonkohler commented:
Looks a lot better, thanks. A few more style nits that I think you can fix by just setting up your IDE autoformatter. Once you change startMaster(), I think we're set.

damonkohler committed:
Adds Context argument to draw() method for OpenGlDrawables. This enables loading assets, etc.

tulku commented:
Thanks for the review Damon. This file has some wrapping inconsistencies that predate out changes. Would like me to re format the file (adding lots of changes to the PR) or just keep the modified lines consistent?

damonkohler commented:
Just the changed lines is great. Sorry for the noise.

damonkohler committed:
Changes use of FrameName back to GraphName now that FrameTransformTree handles GraphNames in a tf2 compatible way.

stonier committed:
Merge pull request #178 from damonkohler/master

New Android library naming scheme and bug fixes.

damonkohler committed:
Merge pull request #181 from damonkohler/next_big_thing

Changes FrameName to GraphName to work with rosjava_core/master

tulku committed:
Merge branch 'master' of https://github.com/rosjava/android_core into choose-network-iface

Conflicts solved in:
	android_10/src/org/ros/android/MasterChooser.java
	android_10/src/org/ros/android/NodeMainExecutorService.java

damonkohler commented:
Ping. Is this ready for another look?

tulku committed:
Move startMaster AsyncTask to NodeExecutorService

Also, merge this branch with master and solved small conflicts.
Fix some more style problems.

tulku commented:
Hi! Sorry for the delay. Between RoboBusiness and traveling I could not get to this. I'll be closing this PR and creating a new one against master. I've already merged master into this branch, so there should be no conflicts.
